binderhub/build.py

```diff
@@ -382,6 +393,13 @@ def submit(self):
                 client.V1EnvVar(name="GIT_CREDENTIAL_ENV", value=self.git_credentials)
             )
 
+        if "privileged" in self.build_capabilities:
```

Comment:
Can we just ask for something like `build_security_context` or something and just pass it on directly through?

Part of the complexity here is that we want this to be not kubernetes specific. However, I think the way to do that is to refactor out the current class to be a KubeBuilder or something, and use traitlets directly. But until then, I think we should pass config through directly as much as possible.

Reply:
Not easily, we'd need to recurse through the JSON dict by copying this KubeSpawner code:
https://github.com/jupyterhub/kubespawner/blob/63aaccc567d03110fb83c19cbdbfdc1a30eb5406/kubespawner/utils.py#L92-L188

Good point about the long-term plan to make this into a traitlets Configurable.... let me see if I can come up with something

Reply:
@yuvipanda I've added a new traitlets based build class in https://github.com/jupyterhub/binderhub/pull/1518